t/runner

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+
+import os
+import requests
+import signal
+import socket
+import subprocess
+import time
+import unittest
+
+
+TESTS_DIR = os.path.dirname(__file__)
+UWSGI_BINARY = os.getenv('UWSGI_BINARY',
+                         os.path.join(TESTS_DIR, '..', 'uwsgi'))
+UWSGI_ADDR = '127.0.0.1'
+UWSGI_PORT = 8000
+UWSGI_HTTP = f'{UWSGI_ADDR}:{UWSGI_PORT}'
+
+
+class BaseTest:
+    """
+    Container class to avoid base test being run
+    """
+
+    class UwsgiServerTest(unittest.TestCase):
+        """
+        Test case with a server instance available on a socket for requests
+        """
+
+        @classmethod
+        def uwsgi_ready(cls):
+            try:
+                s = socket.socket()
+                s.connect((UWSGI_ADDR, UWSGI_PORT, ))
+            except socket.error:
+                return False
+            else:
+                return True
+            finally:
+                s.close()
+
+        @classmethod
+        def setUpClass(cls):
+            # launch server
+            cls.testserver = subprocess.Popen([UWSGI_BINARY,
+                                               '--http-socket', UWSGI_HTTP]
+                                              + cls.ARGS)
+
+            # ensure server is ready
+            retries = 10
+            while not cls.uwsgi_ready() and retries > 0:
+                time.sleep(0.1)
+                retries = retries - 1
+                if retries == 0:
+                    raise RuntimeError('uwsgi test server is not available')
+
+        @classmethod
+        def tearDownClass(cls):
+            cls.testserver.send_signal(signal.SIGTERM)
+            cls.testserver.wait()
+
+
+class StaticTest(BaseTest.UwsgiServerTest):
+
+    ARGS = ['--plugin', 'python3', # provide a request plugin if no embedded request plugin
```

Comment:
`'--plugin', 'python3' ` should not be required and loading the python plugin is failing because it's embeded

Reply:
It would be great of the integration test work with other build configurations, such as builds with no embedded plugins. That's the workaround I've found, otherwise `uwsgi` complains of no loaded request plugin. Do you have a better suggestion? Or this can be conditional.